core/src/com/unciv/ui/worldscreen/Minimap.kt

```diff
@@ -120,13 +123,65 @@ class Minimap(val mapHolder: WorldMapHolder) : Table(){
 class MinimapHolder(mapHolder: WorldMapHolder): Table() {
     val minimap = Minimap(mapHolder)
     val worldScreen = mapHolder.worldScreen
+    private data class ToggleButtonInfo(val actor: Actor, val getIt: ()->Boolean, val setIt: (Boolean)->Unit)
+    private val toggleButtonInfo: EnumMap<MinimapToggleButtons,ToggleButtonInfo> = EnumMap<MinimapToggleButtons,ToggleButtonInfo>(MinimapToggleButtons::class.java)
 
     init {
         add(getToggleIcons()).align(Align.bottom)
         add(getWrappedMinimap())
         pack()
     }
 
+    enum class MinimapToggleButtons {
+        YIELD {
+            override val icon = "Food"
+        },
+        WORKED {
+            override val icon = "Population"
+        },
+        RESOURCES {
+            override val icon = "ResourceIcons/Cattle"
+        };
+        abstract val icon: String
+    }
+
+    // "Api" when external code wants to toggle something together with our buttons
+    fun getToggleButton(which: MinimapToggleButtons): Boolean {
+        val info = toggleButtonInfo[which] ?: return false
+        return info.getIt()
+    }
+    fun setToggleButton(which: MinimapToggleButtons, value: Boolean) {
+        val info = toggleButtonInfo[which] ?: return
+        info.setIt(value)
+        toggleButtonInfo[which]?.actor?.color?.a = if (value) 1f else 0.5f
```

Comment:
Since you already retrieved info 2 lines above, can't you just use that? It'll also get rid of all the ?.

Reply:
Seems true - probably an accident re-merging (to get rid of 1 extra commit removing test code - I mishandled squashing). The val was expressly meant to avoid the cascaded `?.` 's. Embarrassing.